### PR TITLE
Close or Re-open PR with prow commands

### LINKED ISSUE #2839: [prow] Why doesn't /close handle PRs?
I'm trying to close a PR assigned to me. `/close` did not do anything.

https://github.com/kubernetes/test-infra/blob/27540b574f70e4c25847e55de75b2212a18ecbba/prow/plugins/close/close.go#L47

It looks like this is by design. Why?

### PR DESCRIPTION
**Pull request by cblecker**
Fixes #2839.

I think this works? I'm not sure how to test changes like this.

cc: @spxtr as original author

### CODE REVIEW

**Comment by cblecker** [comment]
Does this need a hook bump? IIRC yes, but wanted to check first.

**Comment by spiffxp** [comment]
commit/PR mistitled?

**Comment by cblecker** [comment]
Hahaha.. Yup, it sure is. Fixing..

File: prow/github/client.go [hunk 1]
```diff
@@ -698,6 +699,30 @@ func (c *Client) ReopenIssue(org, repo string, number int) error {
 	return err
 }
 
+// ClosePR closes the existing, open issue provided
```
Comment (spiffxp) [inline prow/github/client.go]: open PR

File: prow/github/client.go [hunk 2]
```diff
@@ -698,6 +699,30 @@ func (c *Client) ReopenIssue(org, repo string, number int) error {
...
+	_, err := c.request(&request{
+		method:      http.MethodPatch,
+		path:        fmt.Sprintf("%s/repos/%s/%s/pull/%d", c.base, org, repo, number),
+		requestBody: map[string]string{"state": "closed"},
+		exitCodes:   []int{200},
+	}, nil)
+	return err
+}
+
+// ReopenPR re-opens the existing, closed issue provided
```
Comment (spiffxp) [inline prow/github/client.go]: closed PR

**Comment by spiffxp** [comment]
@spxtr maybe CODEOWNERS should be using `prow/**` instead of `prow/*`?

File: prow/github/client.go (hunk 1 above)
Comment (cblecker) [inline prow/github/client.go]: Fixed!

**Comment by xiangpengzhao** [comment]
Another associated issue: #2863

File: prow/plugins/close/close.go [hunk 3]
```diff
@@ -60,11 +61,19 @@ func handle(gc githubClient, log *logrus.Entry, ic github.IssueCommentEvent) err
...
 	if !ic.Issue.IsAuthor(commentAuthor) && !ic.Issue.IsAssignee(commentAuthor) {
-		resp := "you can't close an issue unless you authored it or you are assigned to it"
+		resp := "you can't close an issue/PR unless you authored it or you are assigned to it"
 		log.Infof("Commenting \"%s\".", resp)
 		return gc.CreateComment(org, repo, number, plugins.FormatICResponse(ic.Comment, resp))
 	}
 
-	log.Info("Closing issue.")
-	return gc.CloseIssue(org, repo, number)
+	switch ic.Issue.IsPullRequest() {
```
Comment (spxtr) [inline prow/plugins/close/close.go]: I think `if` statements are the appropriate solution for switching on a bool :)

File: prow/plugins/reopen/reopen.go [hunk 4]
```diff
@@ -60,11 +61,19 @@ func handle(gc githubClient, log *logrus.Entry, ic github.IssueCommentEvent) err
...
 	if !ic.Issue.IsAuthor(commentAuthor) && !ic.Issue.IsAssignee(commentAuthor) {
-		resp := "you can't re-open an issue unless you authored it or you are assigned to it"
+		resp := "you can't re-open an issue/PR unless you authored it or you are assigned to it"
 		log.Infof("Commenting \"%s\".", resp)
 		return gc.CreateComment(org, repo, number, plugins.FormatICResponse(ic.Comment, resp))
 	}
 
-	log.Info("Re-opening issue.")
-	return gc.ReopenIssue(org, repo, number)
+	switch ic.Issue.IsPullRequest() {
```
Comment (spxtr) [inline prow/plugins/reopen/reopen.go]: and again

File: prow/plugins/close/close.go (hunk 3 above)
Comment (cblecker) [inline prow/plugins/close/close.go]: Makes sense! Fixing..

**Comment by cblecker** [comment]
Comments addressed, squashed, and bumped hook :)

**Comment by cblecker** [comment]
@spxtr @spiffxp Tests passed. Ready for merge :shipit:

**Pull request merged**